test-framework/common/src/main/java/io/quarkus/test/common/TestResourceManager.java

```diff
@@ -69,8 +111,22 @@ public void init() {
         return ret;
     }
 
+    private void waitForAllFutures(List<Future> startFutures) {
+        for (Future future : startFutures) {
+            try {
+                future.get();
+            } catch (CancellationException | InterruptedException | ExecutionException e) {
+                throw new RuntimeException("Error waiting for test resource future to finish.", e);
+            }
+        }
+    }
+
+    private ExecutorService newExecutor() {
+        return Executors.newWorkStealingPool();
```

Comment:
I would just use a fixed pool sized to the number of tasks

Reply:
I thought of doing that. I even searched to see if there was any shared thread pool that would be indicated to use here. I can change it, but shouldn't we have a limit at least? The first test to run will be the one starting up all the different test resources (that should never be something like 100, i don't think so, but you never know :p)

What do you think? fixed pool size to the number of tasks with a limit to the number of CPUs?

Reply:
I agree limiting it to the number of CPUs is a good idea.

Reply:
This ^ plus maybe a system property to override the default?